docs/cookbook/extending-product-list.md

```diff
@@ -0,0 +1,129 @@
+     * @param \Shopsys\ReadModelBundle\Image\ImageView|null $image
+     */
+    public function __construct(
+        int $id,
+        string $name,
+        ?string $brandName,
+        ?string $shortDescription,
+        string $availability,
+        ProductPrice $sellingPrice,
+        array $flagIds,
+        ProductActionView $action,
+        ?ImageView $image
+    ) {
+        parent::__construct($id, $name, $shortDescription, $availability, $sellingPrice, $flagIds, $action, $image);
+
+        $this->brandName = $brandName;
+    }
+
+    /**
+     * @return string|null
+     */
+    public function getBrandName(): ?string
+    {
+        return $this->brandName;
+    }
+}
+```
+
+### 2. Extend `ListedProductViewFactory` so it returns the new required data
+
+The class is responsible for creating the view object. We need to ensure that the objects is created with proper brand name. We are able to get the brand name from the product entity, so we just need to overwrite `createFromProduct()` method.  
+
+```php
+declare(strict_types=1);
+
+namespace Shopsys\ShopBundle\Model\Product\View;
+
+use Shopsys\FrameworkBundle\Model\Product\Product;
+use Shopsys\ReadModelBundle\Image\ImageView;
+use Shopsys\ReadModelBundle\Product\Action\ProductActionView;
+use Shopsys\ReadModelBundle\Product\Listed\ListedProductViewFactory as BaseListedProductViewFactory;
+use Shopsys\ReadModelBundle\Product\Listed\ListedProductView;
+
+class ListedProductViewFactory extends BaseListedProductViewFactory
+{
+    /**
+     * @param \Shopsys\FrameworkBundle\Model\Product\Product $product
+     * @param \Shopsys\ReadModelBundle\Image\ImageView|null $imageView
+     * @param \Shopsys\ReadModelBundle\Product\Action\ProductActionView $productActionView
+     * @return \Shopsys\ReadModelBundle\Product\Listed\ListedProductView
+     */
+    protected function createFromProduct(Product $product, ?ImageView $imageView, ProductActionView $productActionView): ListedProductView
+    {
+        return new ListedProductView(
+            $product->getId(),
+            $product->getName(),
+            $product->getBrand() ? $product->getBrand()->getName() : null,
+            $product->getShortDescription($this->domain->getId()),
+            $product->getCalculatedAvailability()->getName(),
+            $this->productCachedAttributesFacade->getProductSellingPrice($product),
+            $this->getFlagIdsForProduct($product),
+            $productActionView,
+            $imageView
+        );
+    }
+}
+```
+
+You need to register your new class as an alias for the one from the bundle in `services.yml`:
+
+```yml
+Shopsys\ReadModelBundle\Product\Listed\ListedProductViewFactory: '@Shopsys\ShopBundle\Model\Product\View\ListedProductViewFactory'
+```
+
+*Note: If you want to avoid ["the n+1 problem"](https://stackoverflow.com/questions/97197/what-is-the-n1-selects-problem-in-orm-object-relational-mapping) here,
```

Comment:
We are suggesting to take a look at `Facade` but we are extending `Factory` in this article.

It sort of makes sense, but if you want to keep here this note, you have to instruct the reader what to do next. Or alternatively you can ass a new section **4** that deals with getting data in an optimized way.

Reply:
in such a case, I would rather remove the note completely, as there are too many scenarios and solutions...it seems out of the scope to me